docs/tutorials/getting-started-with-flask.rst

```diff
@@ -0,0 +1,454 @@
+
+.. note::
+    The size of the image reported by Docker is the uncompressed size which is
+    larger than the size of the compressed ``.rock`` file.
+
+Now you're finally ready to run the rock and test your containerised Flask
+application:
+
+.. literalinclude:: code/getting-started-with-flask/task.yaml
+    :language: text
+    :start-after: [docs:docker-run]
+    :end-before: [docs:docker-run-end]
+    :dedent: 2
+
+Use the same ``curl`` command as before to send a request to the Flask
+application's root endpoint which is running inside the container:
+
+.. literalinclude:: code/getting-started-with-flask/task.yaml
+    :language: text
+    :start-after: [docs:curl-flask-rock]
+    :end-before: [docs:curl-flask-rock-end]
+    :dedent: 2
+
+The Flask application should again respond with ``Hello, world!``.
+
+View the application logs
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When deploying your Flask rock, you can always get the application logs via
+``pebble``:
+
+.. literalinclude:: code/getting-started-with-flask/task.yaml
+    :language: text
+    :start-after: [docs:get-logs]
+    :end-before: [docs:get-logs-end]
+    :dedent: 2
+
+As a result, :ref:`pebble_explanation_page` will give you the logs for the
+``flask`` service running inside the container.
+You should expect to see something similar to this:
+
+..  code-block:: text
+    :class: log-snippets
+
+    2024-06-21T03:41:45.077Z [flask] [2024-06-21 03:41:45 +0000] [17] [INFO] Starting gunicorn 22.0.0
+    2024-06-21T03:41:45.077Z [flask] [2024-06-21 03:41:45 +0000] [17] [INFO] Listening at: http://0.0.0.0:8000 (17)
+    2024-06-21T03:41:45.077Z [flask] [2024-06-21 03:41:45 +0000] [17] [INFO] Using worker: sync
+    2024-06-21T03:41:45.078Z [flask] [2024-06-21 03:41:45 +0000] [18] [INFO] Booting worker with pid: 18
+
+You can also choose to follow the logs by using the ``-f`` option with the
+``pebble logs`` command above.
+
+.. important::
+
+    To get the Flask application logs, the container must be running. This is
+    also valid for the remaining sections of this tutorial.
+
+Cleanup
+~~~~~~~
+
+Now you have a fully functional rock for you Flask application! This concludes
+the first part of this tutorial, so you can stop the container and remove the
+respective image for now:
+
+.. literalinclude:: code/getting-started-with-flask/task.yaml
+    :language: bash
+    :start-after: [docs:stop-docker]
+    :end-before: [docs:stop-docker-end]
+    :dedent: 2
+
+Chisel the rock
+===============
+
+This is an optional but recommended step, especially if you're looking to
+deploy your rock into a production environment. With :ref:`chisel_explanation`
+you can produce lean and production-ready rocks by getting rid of all the
+contents that are not needed for your Flask application to run. This results
+in a much smaller rock with a reduced attack surface.
+
+The first step towards chiselling your rock is to ensure you are using a
+``bare`` base.
```

Comment:
It'll be nice to have some explanation or reference of bases linked here

Reply:
Yep.doing it now that the other PR has been merged

Reply:
done in d255831